### PR TITLE
Upgraded SimpleTZ and DualCore projects to CMSIS 6

### PR DESCRIPTION
**Pull request by omarArm**
1) upgraded both the SimpleTZ and DualCore projects to use CMSIS 6.
2) updated the device name for SimpleTZ to be compatible with Cortex_DFP devices, also added a new RTE folder for the new device name
3) added a patch file in the DualCore project for an error free transitioning from CMSIS 5.9.0 to CMSIS 6.0.0

### CODE REVIEW

File: SimpleTrustZone/SimpleTZ.csolution.yml [hunk 1]
```diff
@@ -4,10 +4,12 @@ solution:
   compiler: AC6
 
   packs:
-    - pack: ARM::CMSIS@5.9.0
+    - pack: ARM::CMSIS@6.0.0
```
Comment (JonatanAntoni) [inline SimpleTrustZone/SimpleTZ.csolution.yml]: Do we need to pin version?

**Review by JonatanAntoni — COMMENTED** [review]
Can we add a cron trigger to the build.yml workflow?
In case of not changing the examples for a longer priod might hide issues introduced by changes to used packs.

File: SimpleTrustZone/SimpleTZ.csolution.yml (hunk 1 above)
Comment (omarArm) [reply SimpleTrustZone/SimpleTZ.csolution.yml]: we don't really need to pin to a certain version, that line of code was set to CMSIS@5.9.0, so I just re-pinned it. Will change it to a floating version then

File: DualCore/HelloWorld.csolution.yml [hunk 2]
```diff
@@ -27,3 +27,5 @@ solution:
   projects:
     - project: ./cm0plus/HelloWorld_cm0plus.cproject.yml
     - project: ./cm4/HelloWorld_cm4.cproject.yml
+
```
Comment (RobertRostohar) [inline DualCore/HelloWorld.csolution.yml:30]: Empty new lines and white spaces should not be added at the end of the file. The file should end with a single new line.

File: DualCore/cm4/cm4_cmsis_5_to_6_patch.h [hunk 3]
```diff
@@ -0,0 +1,137 @@
+#ifndef CM4_CMSIS_5_to_6_PATCH
```
Comment (RobertRostohar) [inline DualCore/cm4/cm4_cmsis_5_to_6_patch.h]: How about a different name of this file? Maybe just cmsis_core5.h. We could also consider moving it under RTE/CMSIS directory.

Missing disclaimer at the top.

We could also add a CMSIS version check at the beginning.

For example:
#if defined (__CM_CMSIS_VERSION_MAIN) && (__CM_CMSIS_VERSION_MAIN >= 6)
#define CoreDebug...
...
#endif

File: SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+*/
+
+/*--------------------- Flash Configuration ----------------------------------
+; <h> Flash Configuration
+;   <o0> Flash Base Address <0x0-0xFFFFFFFF:8>
+;   <o1> Flash Size (in Bytes) <0x0-0xFFFFFFFF:8>
+; </h>
+ *----------------------------------------------------------------------------*/
+#define __ROM_BASE      0x00000000
+#define __ROM_SIZE      0x00080000
```
Comment (RobertRostohar) [inline SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct]: Does not match the previous settings in ARMCM33_DSP_FP_TZ.

File: SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+/*--------------------- Embedded RAM Configuration ---------------------------
+; <h> RAM Configuration
+;   <o0> RAM Base Address    <0x0-0xFFFFFFFF:8>
+;   <o1> RAM Size (in Bytes) <0x0-0xFFFFFFFF:8>
+; </h>
+ *----------------------------------------------------------------------------*/
+#define __RAM_BASE      0x20000000
+#define __RAM_SIZE      0x00040000
```
Comment (RobertRostohar) [inline SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct]: Does not match the previous settings in ARMCM33_DSP_FP_TZ.

File: SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+/*--------------------- Stack / Heap Configuration ---------------------------
+; <h> Stack / Heap Configuration
+;   <o0> Stack Size (in Bytes) <0x0-0xFFFFFFFF:8>
+;   <o1> Heap Size (in Bytes) <0x0-0xFFFFFFFF:8>
+; </h>
+ *----------------------------------------------------------------------------*/
+#define __STACK_SIZE    0x00000200
```
Comment (RobertRostohar) [inline SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct]: Does not match the previous settings in ARMCM33_DSP_FP_TZ.

File: SimpleTrustZone/SimpleTZ.csolution.yml [hunk 7]
```diff
@@ -4,10 +4,12 @@ solution:
   compiler: AC6
 
   packs:
-    - pack: ARM::CMSIS@5.9.0
+    - pack: ARM::CMSIS
+    - pack: ARM::CMSIS-RTX
```
Comment (RobertRostohar) [inline SimpleTrustZone/SimpleTZ.csolution.yml]: CMSIS-RTX pack is not used by this solution and should be removed.

**Review by RobertRostohar — CHANGES_REQUESTED** [review]
The old device directory ARMCM33_DSP_FP_TZ should be removed after it contents has been migrated to ARMCM33. See also inline comments. Note that also gcc configuration files existed and have not been migrated.

It would be good also to run the example with FVP in order to verify that it still works as expected.

File: DualCore/cm4/cm4_cmsis_5_to_6_patch.h (hunk 3 above)
Comment (omarArm) [reply DualCore/cm4/cm4_cmsis_5_to_6_patch.h]: I changed the file name for convenience, but the problem with having a CMSIS version check is that the patch is to be used as a preinclude and not a normal include, that's why I cannot really apply this solution. Maybe we can change the idea of applying the patch file in a different pull request

File: DualCore/HelloWorld.csolution.yml (hunk 2 above)
Comment (omarArm) [reply DualCore/HelloWorld.csolution.yml:30]: done

File: SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct (hunk 4 above)
Comment (omarArm) [reply SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct]: done

File: SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct (hunk 6 above)
Comment (omarArm) [reply SimpleTrustZone/CM33_ns/RTE/Device/ARMCM33/ARMCM33_ac6.sct]: done

**Review by JonatanAntoni — APPROVED** [review]

**Review by JonatanAntoni — APPROVED** [review]

File: DualCore/cm4/cm4_cmsis_5_to_6_patch.h (hunk 3 above)
Comment (RobertRostohar) [reply DualCore/cm4/cm4_cmsis_5_to_6_patch.h]: You could include "cmsis_version.h" inside the patch.
See here:
https://github.com/MDK-Packs/LPCXpresso55S69_BSP/pull/1/files#diff-5a0be11fc7542c47f6ef45678048f445a46cc374dc6d7b14108c35db2dec989a

**Pull request merged**